plugins/catalog/src/components/AboutCard/fields.tsx

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  getEntitySourceLocation,
+  RELATION_OWNED_BY,
+  RELATION_PART_OF,
+} from '@backstage/catalog-model';
+import {
+  EntityRefLinks,
+  getEntityRelations,
+  useEntity,
+} from '@backstage/plugin-catalog-react';
+import { JsonArray } from '@backstage/types';
+import { Chip, makeStyles, Typography } from '@material-ui/core';
+import React from 'react';
+import { AboutField } from './AboutField';
+import { LinksGridList } from '../EntityLinksCard/LinksGridList';
+
+const useStyles = makeStyles({
+  description: {
+    wordBreak: 'break-word',
+  },
+});
+
+function getLocationTargetHref(
+  target: string,
+  type: string,
+  entitySourceLocation: {
+    type: string;
+    target: string;
+  },
+): string {
+  if (type === 'url' || target.includes('://')) {
+    return target;
+  }
+
+  const srcLocationUrl =
+    entitySourceLocation.type === 'file'
+      ? `file://${entitySourceLocation.target}`
+      : entitySourceLocation.target;
+
+  if (type === 'file' || entitySourceLocation.type === 'file') {
+    return new URL(target, srcLocationUrl).href;
+  }
+
+  return srcLocationUrl;
+}
+
+/**
+ * The built-in fields to be used individually.
+ *
+ * @public
+ */
+export const AboutCardBuiltInFields = Object.freeze({
+  Description() {
+    const { entity } = useEntity();
+    const classes = useStyles();
+    return (
+      <AboutField label="Description" gridSizes={{ xs: 12 }}>
+        <Typography variant="body2" paragraph className={classes.description}>
+          {entity?.metadata?.description || 'No description'}
+        </Typography>
+      </AboutField>
+    );
+  },
+  Owner() {
+    const { entity } = useEntity();
+    const ownedByRelations = getEntityRelations(entity, RELATION_OWNED_BY);
+    return (
+      <AboutField
+        label="Owner"
+        value="No Owner"
+        gridSizes={{ xs: 12, sm: 6, lg: 4 }}
+      >
+        {ownedByRelations.length > 0 && (
+          <EntityRefLinks entityRefs={ownedByRelations} defaultKind="group" />
+        )}
+      </AboutField>
+    );
+  },
+  Domain() {
+    const { entity } = useEntity();
+    const isSystem = entity.kind.toLocaleLowerCase('en-US') === 'system';
```

Comment:
I don't think you should do kind checking (neither here nor in the other places). Just look for the relation type + target kind, and if that relation exists, render it. People extend their catalogs with custom kinds all the time so you won't be able to know the proper list here beforehand. Plus it gets hard to maintain, and we're bound to forget places like this if we add more kinds ourselves to the core model :)

Reply:
@freben Thanks for the insights! I simply kept the original logic of those fields and didn't really think much of it 😂 I will address your comments later.

Reply:
Hi @freben, I implemented your suggestions and then realized that removing the kind checks may result in undesired `domain` or `system` fields showing up in the about card of built-in entities.

After the change, it would look like this:
```typescript
<AboutField
  label="Domain"
  value="No Domain"
  gridSizes={{ xs: 12, sm: 6, lg: 4 }}
>
  {partOfDomainRelations.length > 0 && (
    <EntityRefLinks
      entityRefs={partOfDomainRelations}
      defaultKind="domain"
    />
  )}
</AboutField>
```
By default, a `No Domain` field will be rendered if no such relation, and it will apply to all entities, including the built-in entities which actually don't accept a `domain` field. e.g. For the `location` kind, it looks like this:
<img width="791" alt="image" src="https://user-images.githubusercontent.com/9165473/185786657-5fc0273e-0170-4f13-aacb-b7ea41395b3a.png">

This is a bit confusing because there is no way to set such fields for those entities. We can have another "kind check" to filter out built-in entities which are not qualified for setting `domain` or `system`, but I think it goes against our intention here. 

I'm in the dilemma and need your advice here 🙏